src/elements/RegisterModelPopUp/RegisterModelPopUp.js

```diff
@@ -0,0 +1,167 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { useLocation } from 'react-router-dom'
+import { createForm } from 'final-form'
+import { Form } from 'react-final-form'
+import arrayMutators from 'final-form-arrays'
+import { v4 as uuidv4 } from 'uuid'
+
+import { Button, Modal, FormChipCell, FormInput, FormTextarea } from 'igz-controls/components'
+
+import { getChipOptions } from '../../utils/getChipOptions'
+import { convertChipsData } from '../../utils/convertChipsData'
+import { setFieldState } from 'igz-controls/utils/form.util'
+import { useModalBlockHistory } from '../../hooks/useModalBlockHistory.hook'
+import { MODAL_SM, SECONDARY_BUTTON, TERTIARY_BUTTON } from 'igz-controls/constants'
+import notificationActions from '../../actions/notification'
+
+import artifactApi from '../../api/artifacts-api'
+
+import './registerModelPopUp.scss'
+
+function RegisterModelPopUp({ filtersStore, isOpen, onResolve, projectName, refresh, setNotification }) {
```

Comment:
use useSelector hook to connect to the filtersStore, and use useDispatch for actions